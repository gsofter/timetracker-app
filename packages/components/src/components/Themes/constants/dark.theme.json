{
	"name": "dark",
	"type": "dark",
	"colors": {
		"primaryColor": "#1890ff",
		"app.background": "#fff",
		"app.headerHeight": "48px",
		"list.dropBackground": "#383b3d",
		"sideBarTitle.foreground": "#bbbbbb",
		"popover.background": "#1f1f1f",
		"popover.customize.border.color": "#3a3a3a",
		"body.background": "#000000",
		"component.background": "#141414",
		"layout.body.background": "#000000",
		"layout.header.background": "#1f1f1f",
		"layout.trigger.background": "#262626"
	},
	"tokenColors": [
		{
			"scope": "header",
			"settings": {
				"foreground": "#000080"
			}
		},
		{
			"scope": "text",
			"settings": {
				"foreground": "rgba(255,255,255,0.85)"
			}
		},
		{
			"scope": "text.secondary",
			"settings": {
				"foreground": "rgba(255,255,255,0.65)"
			}
		}
	]
}