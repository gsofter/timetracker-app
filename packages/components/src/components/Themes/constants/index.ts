export * from './actionTypes';
export { default as COLORS } from './colors';
export { default as THEMES } from './themes';
