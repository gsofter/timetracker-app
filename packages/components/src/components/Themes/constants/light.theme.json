{	
	"name": "light",
	"type": "light",
	"colors": {
		"primaryColor": "#1890ff",
		"app.background": "#000",
		"app.headerHeight": "48px",
		"list.dropBackground": "#383b3d",
		"sideBarTitle.foreground": "#bbbbbb",
		"popover.background": "#1f1f1f",
		"popover.customize.border.color": "#3a3a3a",
		"body.background": "#000000",
		"component.background": "#141414",
		"layout.body.background": "#000000",
		"layout.header.background": "#1f1f1f",
		"layout.trigger.background": "#262626"
	},
	"tokenColors": [
		{
			"scope": "header",
			"settings": {
				"foreground": "#262626"
			}
		},
		{
			"scope": "text",
			"settings": {
				"foreground": "#595959"
			}
		},
		{
			"scope": "text.secondary",
			"settings": {
				"foreground": "#8c8c8c"
			}
		}
	]
}