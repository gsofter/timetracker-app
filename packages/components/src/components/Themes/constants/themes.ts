import { IThemeSettings } from '../interfaces';
const lightSettings: IThemeSettings = require('./light.theme.json');
const darkSettings: IThemeSettings = require('./dark.theme.json');
const hcSettings: IThemeSettings = require('./hc.theme.json');

export default [
  {
    name: 'light',
    className: '',
    text: 'Light',
    defaultSettings: lightSettings,
  },
  {
    name: 'dark',
    className: 'vs-dark',
    text: 'Dark',
    defaultSettings: darkSettings,
  },
  {
    name: 'hc',
    className: 'hc-black',
    text: 'High Contrast',
    defaultSettings: hcSettings,
  },
];
