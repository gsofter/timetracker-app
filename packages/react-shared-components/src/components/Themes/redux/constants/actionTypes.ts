export const THEME_ACTION_TYPES = {
    SET_VALUE: 'theme/SET_VALUE',
    BULK_UPDATE: 'theme/BULK_UPDATE',
    LOAD_THEME: 'theme/LOAD_THEME',
    SET_THEME: 'theme/SET_THEME',
    SET_SETTING: 'theme/SET_SETTING',
};
