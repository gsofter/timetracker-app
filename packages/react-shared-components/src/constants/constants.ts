


export const CHANGE_SETTINGS_ACTION = 'settings/changeSetting';