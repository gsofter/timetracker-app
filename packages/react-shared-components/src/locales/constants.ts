export const LOCALES = {
    EN_US: 'en-US',
    PT_BR: 'pt-BR',
    ZH_CN: 'zh-CN',
    ZH_TW: 'zh-TW',
    ID_ID: 'id-ID',
  };