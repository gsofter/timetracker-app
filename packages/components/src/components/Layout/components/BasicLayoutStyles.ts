import { getTokenColorsScopeSettings } from '../../Themes/redux/utils';
import { IThemeState } from '../../Themes/redux';
import { GlobalThemeSettings, LocalThemeSettings } from './default-styles';
import { Property, Properties } from 'csstype';


export const styleSheet = {
	basicLayout: ({ theme, prefixCls }: { theme: IThemeState, prefixCls: string }) => {
		const { defaultSettings: { colors: { primaryColor }, } } = theme;
		const { antPrefix } = GlobalThemeSettings;
		// less variables
		const basicLayoutPrefixCls = `${antPrefix}-pro-basicLayout`;
		const proLayoutHeaderHeight = LocalThemeSettings.proLayoutHeaderHeight;
		return {
			// @sri modified to work with Fela
			display: 'flex',
			width: '100%',
			'min-height': '100%',
			[`& .${basicLayoutPrefixCls}`]: {
			// BFC
			display: 'flex',
			'flex-direction': 'column',
			width: '100%',
			'min-height': '100%',

				[`& .${antPrefix}-layout-header`]: {
					[`&.${antPrefix}-pro-fixed-header`]: {
						position: 'fixed',
						top: '0',
					}
				},

				'&-content': {
					position: 'relative',
					margin: '24px',

					[`& .${antPrefix}-pro-page-container`]: {
						margin: '-24px -24px 0',
					},

					'&-disable-margin': {
						margin: '0',

						[`& .${antPrefix}-pro-page-container`]: {
							margin: '0',
						}
					},
					[`> .${antPrefix}-layout`]: {
						'max-height': '100%',
					}
				},

				// children should support fixed
				[`& .${basicLayoutPrefixCls}-is-children.${basicLayoutPrefixCls}-fix-siderbar`]: {
					height: '100vh',
					overflow: 'hidden',
					transform: 'rotate(0)',
				},

				[`& .${basicLayoutPrefixCls}-has-header`]: {
					// tech-page-container
					'& .tech-page-container': {
						height: `calc(100vh - ${proLayoutHeaderHeight})`,
					},
					[`& .${basicLayoutPrefixCls}-is-children.${basicLayoutPrefixCls}-has-header`]: {
						'& .tech-page-container': {
							height: `calc(100vh - ${proLayoutHeaderHeight} - ${proLayoutHeaderHeight})`,
						},
						[`& .${basicLayoutPrefixCls}-is-children`]: {
							'min-height': `calc(100vh - ${proLayoutHeaderHeight})`,
							[`&.${basicLayoutPrefixCls}-fix-siderbar`]: {
								height: `calc(100vh - ${proLayoutHeaderHeight})`,
							}
						}
					}
				}
			}
		};
	},
};
