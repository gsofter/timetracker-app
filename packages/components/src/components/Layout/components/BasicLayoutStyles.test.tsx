import 'raf/polyfill'
import * as React from 'react';
import { createSnapshot } from 'jest-react-fela';
import { createComponent } from 'react-fela';
import { THEMES } from '../../Themes/constants/index'

import { styleSheet } from './BasicLayoutStyles';
import 'jest';


const rule = (props) => styleSheet.basicLayout({
  theme: THEMES[0].defaultSettings
});
describe('Test styles', () => {

  it('should return a Component', () => {


    const Component = createComponent(rule)

    expect(createSnapshot(<Component className='ant-pro' />)).toMatchSnapshot()
  })

})


