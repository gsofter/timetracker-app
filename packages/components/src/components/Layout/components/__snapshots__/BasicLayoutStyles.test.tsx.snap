// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test styles should return a Component 1`] = `
".a {
  display: flex;
}
.b {
  width: 100%;
}
.c {
  min-height: 100%;
}
.d .ant-pro-basicLayout {
  display: flex;
}
.e .ant-pro-basicLayout {
  flex-direction: column;
}
.f .ant-pro-basicLayout {
  width: 100%;
}
.g .ant-pro-basicLayout {
  min-height: 100%;
}
.h .ant-pro-basicLayout .ant-layout-header.ant-pro-fixed-header {
  position: fixed;
}
.i .ant-pro-basicLayout .ant-layout-header.ant-pro-fixed-header {
  top: 0;
}
.j .ant-pro-basicLayout-content {
  position: relative;
}
.k .ant-pro-basicLayout-content {
  margin: 24px;
}
.l .ant-pro-basicLayout-content .ant-pro-page-container {
  margin: -24px -24px 0;
}
.m .ant-pro-basicLayout-content-disable-margin {
  margin: 0;
}
.n .ant-pro-basicLayout-content-disable-margin .ant-pro-page-container {
  margin: 0;
}
.o .ant-pro-basicLayout-content > .ant-layout {
  max-height: 100%;
}
.p
  .ant-pro-basicLayout
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-fix-siderbar {
  height: 100vh;
}
.q
  .ant-pro-basicLayout
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-fix-siderbar {
  overflow: hidden;
}
.r
  .ant-pro-basicLayout
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-fix-siderbar {
  transform: rotate(0);
}
.s .ant-pro-basicLayout .ant-pro-basicLayout-has-header .tech-page-container {
  height: calc(100vh - 48px);
}
.t
  .ant-pro-basicLayout
  .ant-pro-basicLayout-has-header
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-has-header
  .tech-page-container {
  height: calc(100vh - 48px - 48px);
}
.u
  .ant-pro-basicLayout
  .ant-pro-basicLayout-has-header
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-has-header
  .ant-pro-basicLayout-is-children {
  min-height: calc(100vh - 48px);
}
.v
  .ant-pro-basicLayout
  .ant-pro-basicLayout-has-header
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-has-header
  .ant-pro-basicLayout-is-children.ant-pro-basicLayout-fix-siderbar {
  height: calc(100vh - 48px);
}


<div className=ant-pro a b c d e f g h i j k l m n o p q r s t u v />;
"
`;
